File: src/ios/OWMobilePlugin.m
/********* OWMobilePlugin.m Cordova Plugin Implementation *******/

#import <Cordova/CDV.h>

@interface OWMobilePlugin : CDVPlugin {
  // Member variables go here.
}
- (void)login:(CDVInvokedUrlCommand*)command;
- (void)logout:(CDVInvokedUrlCommand*)command;
- (void)register:(CDVInvokedUrlCommand*)command;
@end

@implementation OWMobilePlugin

- (void)login:(CDVInvokedUrlCommand*)command
{
  // TODO: Integrate OneWelcome iOS SDK login here

    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Login triggered from native iOS!"];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)logout:(CDVInvokedUrlCommand*)command
{
  // TODO: Integrate OneWelcome iOS SDK logout here

    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Logout triggered from native iOS!"];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

- (void)register:(CDVInvokedUrlCommand*)command
{
  // TODO: Integrate OneWelcome iOS SDK register here

    CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK messageAsString:@"Register triggered from native iOS!"];
    [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

@end

File: src/android/OWMobilePlugin.java
package cordova-ow-plugin;

import org.apache.cordova.CallbackContext;
import org.apache.cordova.CordovaPlugin;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;


public class OWMobilePlugin extends CordovaPlugin {

  @Override
  public boolean execute(String action, JSONArray args, CallbackContext callbackContext) throws JSONException {
    if ("login".equals(action)) {
      this.login(callbackContext);
      return true;
    }
    if ("logout".equals(action)) {
      this.logout(callbackContext);
      return true;
    }
    if ("register".equals(action)) {
      this.register(callbackContext);
      return true;
    }
    return false;
  }

  private void login(CallbackContext callbackContext) {
    // TODO: Integrate OneWelcome Android SDK login here
    callbackContext.success("Login triggered from native Android!");
  }

  private void logout(CallbackContext callbackContext) {
    // TODO: Integrate OneWelcome Android SDK login here
    callbackContext.success("Logout triggered from native Android!");
  }

  private void register(CallbackContext callbackContext) {
    // TODO: Integrate OneWelcome Android SDK login here
    callbackContext.success("Register triggered from native Android!");
  }
}
